RecoLocalMuon/CSCSegment/src/CSCSegAlgoTC.cc

```diff
@@ -762,17 +503,17 @@ bool CSCSegAlgoTC::isSegmentGood(std::vector<ChamberHitContainer>::iterator seg,
   return true;
 }
 
-void CSCSegAlgoTC::flagHitsAsUsed(std::vector<ChamberHitContainer>::iterator seg, const ChamberHitContainer& rechitsInChamber, 
-                                  BoolContainer& used) const {
+void CSCSegAlgoTC::flagHitsAsUsed(std::vector<CSCSegFit*>::iterator seg, 
+      const ChamberHitContainer& rechitsInChamber, BoolContainer& used) const {
 
   // Flag hits on segment as used
 
   ChamberHitContainerCIt ib = rechitsInChamber.begin();
+  ChamberHitContainer hits = (*seg)->hits();
```

Comment:
here and above it's becoming unclear why copies are needed ... this actually originates in ``` CSCSetOfHits hits(void) const { return hits_; } ```